src/traces/table/plot.js

```diff
@@ -0,0 +1,82 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var table = require('./render');
+
+module.exports = function plot(gd, cdTable) {
+
+    var fullLayout = gd._fullLayout;
+    var svg = fullLayout._paper;
+    var root = fullLayout._paperdiv;
+
+    var gdColumns = {};
+    var gdColumnsOriginalOrder = {};
+
+    var size = fullLayout._size;
+
+    cdTable.forEach(function(d, i) {
+        gdColumns[i] = gd.data[i].header.values.map(function(d) {return d[0];});
```

Comment:
Stuff like is perfect for `calc.js`. You could stash those `gdColumns` arrays in `gd.calcdata` and those will only get recomputed when array attributes are updated.